config/routes.rb

```diff
@@ -0,0 +1,21 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+
+  root 'homepages#index'
+  resources :drivers do
+    resources :trips, only: [:index]
+  end
+
+  resources :passengers do
+    resources :trips, only: [:index, :new, :create]
```

Comment:
Do you really need the new trip form?  You can just automatically create a trip from the link.